README.md

```diff
@@ -44,6 +44,40 @@ describe("Expect.Operators", () => {
 
 ```
 
+`testAsync` example with ReactTestUtils `actAsync` for effectfull component assertion:
+
+```reason
+open Jest;
+
+describe("ReactTestUtils actAsync", () => {
+  open Expect;
+  open Js.Promise;
+  open ReactTestUtils;
+
+  let container = ref(None);
+  beforeEach(prepareContainer(container));
+  afterEach(cleanupContainer(container));
+
+  let effectfulComponent = <p>{"Hello async Jest" |> React.string}</p>
+
+  // use testAsync to get a `finish` callback for final assertion
+  testAsync("actAsync", finish => {
+    let container = getContainer(container);
+    // actAsync returns a promise for all effects evaluation
+    actAsync(_ =>
+      // Convert ReactDOMRe.render to a promise
+      resolve(ReactDOMRe.render(effectfulComponent, container))
+    ) |> then_(_ =>
+      // Call finish with assertion on final container
+      finish(expect(
+        container
+        ->DOM.findBySelectorAndTextContent("p", "Hello async Jest")
+        ->Belt.Option.isSome) |> toBe(true)) |> resolve
+    ) |> ignore;
+  });
```

Comment:
It would be more appropriate to use `testPromise` here, I think. Then you should be able to do:

```suggestion
  testPromise("actAsync", () => {
    let container = getContainer(container);
    // actAsync returns a promise for all effects evaluation
    actAsync(_ =>
      // Convert ReactDOMRe.render to a promise
      resolve(ReactDOMRe.render(effectfulComponent, container))
    ) |> then_(_ =>
      // Call finish with assertion on final container
      expect(
        container
        ->DOM.findBySelectorAndTextContent("p", "Hello async Jest")
        ->Belt.Option.isSome) |> toBe(true) |> resolve
    );
  });
```

Slightly simpler and much safer.

Reply:
That looks much better, unfortunately that does not seems to work with actAsync, here is the error message:

```
    TypeError: Cannot read property 'then' of undefined

      at Object.<anonymous> (node_modules/@glennsl/bs-jest/src/jest.bs.js:249:47)
```

Could it be caused by the actAsync binding implementation ( https://github.com/reasonml/reason-react/blob/5ae20e8a6245a584a8f452bf797a2ac26c09d873/src/ReactTestUtils.re#L22-L28 ) ?

Reply:
For some reason, it seems like actAsync resulting promise can only be resolved once. e.g. `actAsync(_ => () |> resolve) |> then_(_ => () |> resolve)` is undefined.

Reply:
Huh, that's weird. They do seem to [return a broken thenable](https://github.com/facebook/react/blob/30b47103d4354d9187dc0f1fb804855a5208ca9f/packages/react-dom/src/test-utils/ReactTestUtilsAct.js#L141-L175). I can't imagine why they would do that without documenting it. Definitely seems like a bug, so perhaps it would be a good idea to open an issue on the react repo?